### ISSUE TITLE
fread verbose message about quoting is fragmented

### ISSUE THREAD

**Issue by MichaelChirico** [Rdatatable/data.table#6543]
> part of above message, perhaps difficult to translate out of context.

Good spot, there's a similar fragment below. It's a bit hard to pull apart exactly how we'd make this friendlier. I think the best bet is to make each fragment a standalone sentence. I'll want to read more carefully how exactly those show up in the verbose output. Filing as a separate issue so it's not lost, there's already a lot going on here.

_Originally posted by @MichaelChirico in https://github.com/Rdatatable/data.table/issues/6503#issuecomment-2379797767_
            
https://github.com/Rdatatable/data.table/blob/37347268b043345c5ef448b7c1163f7bb02395c7/src/fread.c#L1713

https://github.com/Rdatatable/data.table/blob/37347268b043345c5ef448b7c1163f7bb02395c7/src/fread.c#L1763

**Comment by venom1204** [comment]
hi @MichaelChirico can i work on this 

below is my proposed Solution

DTPRINT(_("Detected %d fields per line in the file, using quote rule %d (defines how quotes are handled).\n"), topNumFields, quoteRule);

DTPRINT(_("Processed %d lines with %d fields per line, using quote rule %d (controls the handling of quoted text).\n"), topNumLines, topNumFields, topQuoteRule);

**Comment by MichaelChirico** [comment]
Thanks for offering to take this up!

As noted in my issue, "I'll want to read more carefully how exactly those show up in the verbose output". I can't evaluate your proposal in isolation from how the messages actually appear to the end user.

Please include some sample code that produces the messages so we can compare side-by-side.

**Comment by venom1204** [comment]
[sample.csv](https://github.com/user-attachments/files/18565048/sample.csv)

i created a csv file 

sample_data <- "a,b,c\n1,2,3\n4,5,6\n7,8,9"
write.csv(sample_data, "sample.csv", row.names = FALSE, quote = FALSE)

# Read the file using fread with verbose=TRUE
library(data.table)
result <- fread("sample.csv", verbose = TRUE)

and then checked it for both the current message and the message i changed 

and it was showing 

sep=',' with 4 lines of 3 fields using quote rule 0 [with the current message]


and 

sep=','
Processed 4 lines with 3 fields per line, using quote rule 0 (controls the handling of quoted text).  [with the changed one ]

@MichaelChirico 
The message is now split into two lines for better readability.
"Processed" is added to clarify what the numbers represent.
"per line" is added to make it clear that the field count is per line, not total.
An explanation of what the quote rule means is added in parentheses.


however only one message is showing up not the other one i am working on it.

**Comment by MichaelChirico** [comment]
Please copy-paste the complete `verbose=TRUE` output in your update

**Comment by venom1204** [comment]
> library(data.table)
result <- fread("sample.csv", verbose = TRUE)
  OpenMP version (_OPENMP)       201511
  omp_get_num_procs()            16
  R_DATATABLE_NUM_PROCS_PERCENT  unset (default 50)
  R_DATATABLE_NUM_THREADS        unset
  R_DATATABLE_THROTTLE           unset (default 1024)
  omp_get_thread_limit()         2147483647
  omp_get_max_threads()          16
  OMP_THREAD_LIMIT               unset
  OMP_NUM_THREADS                unset
  RestoreAfterFork               true
  data.table is using 8 threads with throttle==1024. See ?setDTthreads.
freadR.c has been passed a filename: sample.csv
[01] Check arguments
  Using 8 threads (omp_get_max_threads()=16, nth=8)
  NAstrings = [<<NA>>]
  None of the NAstrings look like numbers.
  show progress = 1
  0/1 column will be read as integer
  Y/N column will be read as character
[02] Opening the file
  Opening file sample.csv
  File opened, size = 26 bytes.
  Memory mapped ok
[03] Detect and skip BOM
[04] Arrange mmap to be \0 terminated
  \n has been found in the input and different lines can end with different line endings (e.g. mixed \n and \r\n in one file). This is common and ideal.
[05] Skipping initial rows if needed
  Positioned on line 1 starting: <<x>>
[06] Detect separator, quoting rule, and ncolumns
  Detecting sep automatically ...
  sep=','
Processed 4 lines with 3 fields per line, using quote rule 0 (controls the handling of quoted text).
  Detected 3 columns on line 1. This line is either column names or first data row. Line starts as: <<a,b,c>>
  Quote rule picked = 0
  fill=false and the most number of columns found is 3
[07] Detect column types, dec, good nrow estimate and whether first row is column names
  sep=',' so dec set to '.'
  Number of sampling jump points = 1 because (23 bytes from row 1 to eof) / (2 * 23 jump0size) == 0
  Type codes (jump 000)    : 777  Quote rule 0
  'header' determined to be true due to column 1 containing a string on row 1 and a lower type (int32) in the rest of the 3 sample rows
  All rows were sampled since file is small so we know nrow=3 exactly
[08] Assign column names
[09] Apply user overrides on column types
  After 0 type and 0 drop user overrides : 777
[10] Allocate memory for the datatable
  Allocating 3 column slots (3 - 0 dropped) with 3 rows
[11] Read the data
  jumps=[0..1), chunk_size=1048576, total_size=17
Read 3 rows x 3 columns from 26 bytes file in 00:00.000 wall clock time
[12] Finalizing the datatable
  Type counts:
         3 : int32     '7'
=============================
   0.000s ( 30%) Memory map 0.000GB file
   0.000s ( 52%) sep=',' ncol=3 and header detection
   0.000s (  4%) Column type detection using 3 sample rows
   0.000s (  5%) Allocation of 3 rows x 3 cols (0.000GB) of which 3 (100%) rows used
   0.000s (  9%) Reading 1 chunks (0 swept) of 1.000MB (each chunk 3 rows) using 1 threads
   +    0.000s (  1%) Parse to row-major thread buffers (grown 0 times)
   +    0.000s (  0%) Transpose
   +    0.000s (  8%) Waiting
   0.000s (  0%) Rereading 0 columns due to out-of-sample type exceptions
   0.000s        Total
>

**Comment by MichaelChirico** [comment]
Thanks, please also include the full output on `master` without your change

**Comment by venom1204** [comment]
> library(data.table)
result <- fread("sample.csv", verbose = TRUE)
data.table 1.16.99 IN DEVELOPMENT built 2025-01-29 00:19:44 UTC using 8 threads (see ?getDTthreads).  Latest news: r-datatable.com
  OpenMP version (_OPENMP)       201511
  omp_get_num_procs()            16
  R_DATATABLE_NUM_PROCS_PERCENT  unset (default 50)
  R_DATATABLE_NUM_THREADS        unset
  R_DATATABLE_THROTTLE           unset (default 1024)
  omp_get_thread_limit()         2147483647
  omp_get_max_threads()          16
  OMP_THREAD_LIMIT               unset
  OMP_NUM_THREADS                unset
  RestoreAfterFork               true
  data.table is using 8 threads with throttle==1024. See ?setDTthreads.
freadR.c has been passed a filename: sample.csv
[01] Check arguments
  Using 8 threads (omp_get_max_threads()=16, nth=8)
  NAstrings = [<<NA>>]
  None of the NAstrings look like numbers.
  show progress = 1
  0/1 column will be read as integer
  Y/N column will be read as character
[02] Opening the file
  Opening file sample.csv
  File opened, size = 26 bytes.
  Memory mapped ok
[03] Detect and skip BOM
[04] Arrange mmap to be \0 terminated
  \n has been found in the input and different lines can end with different line endings (e.g. mixed \n and \r\n in one file). This is common and ideal.
[05] Skipping initial rows if needed
  Positioned on line 1 starting: <<x>>
[06] Detect separator, quoting rule, and ncolumns
  Detecting sep automatically ...
  sep=','  with 4 lines of 3 fields using quote rule 0
  Detected 3 columns on line 1. This line is either column names or first data row. Line starts as: <<a,b,c>>
  Quote rule picked = 0
  fill=false and the most number of columns found is 3
[07] Detect column types, dec, good nrow estimate and whether first row is column names
  sep=',' so dec set to '.'
  Number of sampling jump points = 1 because (23 bytes from row 1 to eof) / (2 * 23 jump0size) == 0
  Type codes (jump 000)    : 777  Quote rule 0
  'header' determined to be true due to column 1 containing a string on row 1 and a lower type (int32) in the rest of the 3 sample rows
  All rows were sampled since file is small so we know nrow=3 exactly
[08] Assign column names
[09] Apply user overrides on column types
  After 0 type and 0 drop user overrides : 777
[10] Allocate memory for the datatable
  Allocating 3 column slots (3 - 0 dropped) with 3 rows
[11] Read the data
  jumps=[0..1), chunk_size=1048576, total_size=17
Read 3 rows x 3 columns from 26 bytes file in 00:00.000 wall clock time
[12] Finalizing the datatable
  Type counts:
         3 : int32     '7'
=============================
   0.000s ( 29%) Memory map 0.000GB file
   0.000s ( 48%) sep=',' ncol=3 and header detection
   0.000s (  5%) Column type detection using 3 sample rows
   0.000s (  5%) Allocation of 3 rows x 3 cols (0.000GB) of which 3 (100%) rows used
   0.000s ( 14%) Reading 1 chunks (0 swept) of 1.000MB (each chunk 3 rows) using 1 threads
   +    0.000s (  1%) Parse to row-major thread buffers (grown 0 times)
   +    0.000s (  1%) Transpose
   +    0.000s ( 12%) Waiting
   0.000s (  0%) Rereading 0 columns due to out-of-sample type exceptions
   0.000s        Total
> 

this above messgae is the message coming without any changes 


```diff
- sep=',' with 4 lines of 3 fields using quote rule 0
+ sep=','
+ Processed 4 lines with 3 fields per line, using quote rule 0 (controls the handling of quoted text).

```

**Comment by MichaelChirico** [comment]
@venom1204 thank you for your persistence on this issue. Please be aware that this is a volunteer project; there are no SLAs/SLOs about response times. We get to issues according to availability & priority; right now the priority is the 1.17.0 release.

**Comment by venom1204** [comment]
Thank you for the clarification, and I truly appreciate the time and effort you and the team put into this project. I completely understand that priorities need to be managed, and I respect the commitment to the 1.17.0 release. I'll be patient and look forward to any updates when time permits. Thanks again for all your hard work.

**Comment by MichaelChirico** [comment]
Thanks for following up! I think the easiest thing to do is just combine these messages to call `DTPRINT()` only once. I don't think we need to change the exposition or newlines at all.

**Comment by venom1204** [comment]
Thanx for the reply! Could you please verify—if I am not wrong, you are asking to merge these two DTPRINT statements?

merging this 
```{c}
 DTPRINT((unsigned)sep<32 ? "  sep=%#02x" : "  sep='%c'", sep); // # notranslate
```
with this one  
```{c}
 DTPRINT(_("  with %d lines of %d fields using quote rule %d\n"), topNumLines, topNumFields, topQuoteRule);
```
and 
this 
```{c}
  DTPRINT((unsigned)sep<32 ? "  sep=%#02x" : "  sep='%c'", sep); // # notranslate
```
with 
```{c}
  DTPRINT(_("  with %d fields using quote rule %d\n"), topNumFields, quoteRule);
```

Just want to confirm before making the changes. Thanks!

**Comment by MichaelChirico** [comment]
yes that looks right
